File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Ai/AiChatResult.ts
import { ApiObjectPath } from "../Code/ApiObjectPath";
import { AiChatMessage } from "./AiChatMessage";


export class AiChatResult implements AutoRest.SdkExplorer.Interface.AiChatResult {

    functionPath?: ApiObjectPath;
    message?: AiChatMessage;

    constructor(data: AutoRest.SdkExplorer.Interface.AiChatResult) {
        if(data.message)
            this.message = new AiChatMessage(data.message);
        if (data.functionPath)
            this.functionPath = new ApiObjectPath(data.functionPath);
    }
}

File: src/AutoRest.SdkExplorer/Model/OpenAi/AiChatResult.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using AutoRest.SdkExplorer.Model.Code;

namespace AutoRest.SdkExplorer.Model.OpenAi
{
    public class AiChatResult
    {
        public ApiObjectPath? FunctionPath { get; set; }
        public AiChatMessage Message { get; set; }

        public AiChatResult(AiChatMessage msg, ApiObjectPath? functionpath = null)
        {
            this.Message = msg;
        }
    }
}
